p2p/host.go

```diff
@@ -34,88 +34,85 @@ import (
 	ma "github.com/multiformats/go-multiaddr"
 )
 
-const DefaultProtocolId = protocol.ID("/gossamer/dot/0")
-
 // host wraps libp2p host with host services and information
 type host struct {
-	ctx         context.Context
-	h           libp2phost.Host
-	dht         *kaddht.IpfsDHT
-	bootnodes   []peer.AddrInfo
-	noBootstrap bool
-	noGossip    bool
-	noMdns      bool
-	noStatus    bool
-	address     ma.Multiaddr
-	protocolId  protocol.ID
+	ctx        context.Context
+	h          libp2phost.Host
+	dht        *kaddht.IpfsDHT
+	bootnodes  []peer.AddrInfo
+	protocolId protocol.ID
 }
 
 // newHost creates a host wrapper with a new libp2p host instance
 func newHost(ctx context.Context, cfg *Config) (*host, error) {
-	opts, err := cfg.buildOpts()
+
+	// use "p2p" for multiaddress format
+	ma.SwapToP2pMultiaddrs()
+
+	// create multiaddress (no host id yet)
+	addr, err := ma.NewMultiaddr(fmt.Sprintf("/ip4/0.0.0.0/tcp/%d", cfg.Port))
 	if err != nil {
 		return nil, err
 	}
 
+	// set connection manager
+	cm := &ConnManager{}
+
+	// set libp2p host options
+	opts := []libp2p.Option{
+		libp2p.ListenAddrs(addr),
+		libp2p.DisableRelay(),
+		libp2p.Identity(cfg.privateKey),
+		libp2p.NATPortMap(),
+		libp2p.Ping(true),
+		libp2p.ConnectionManager(cm),
+	}
+
+	// create libp2p host instance
 	h, err := libp2p.New(ctx, opts...)
 	if err != nil {
 		return nil, err
 	}
 
-	// use default protocol if none provided
-	protocolId := protocol.ID(cfg.ProtocolId)
-	if protocolId == "" {
-		protocolId = DefaultProtocolId
-	}
-
-	// create new datastore and DHT
-	dstore := dsync.MutexWrap(ds.NewMapDatastore())
-	dht := kaddht.NewDHT(ctx, h, dstore)
+	// create DHT service
+	dht := kaddht.NewDHT(ctx, h, sync.MutexWrap(ds.NewMapDatastore()))
 
-	// wrap host and DHT with routed host so that we can look up peers in DHT
+	// wrap host and DHT service with routed host
 	h = rhost.Wrap(h, dht)
 
-	// use "p2p" for multiaddress format
-	ma.SwapToP2pMultiaddrs()
```

Comment:
Don't we need this?

Reply:
yes, just moved it to the top of `newHost`